internals/osutil/mkdirallchown.go

```diff
@@ -31,16 +31,23 @@ var mu sync.Mutex
 
 // MkdirAllChown is like os.MkdirAll but it calls os.Chown on any
 // directories it creates.
-func MkdirAllChown(path string, perm os.FileMode, uid sys.UserID, gid sys.GroupID) error {
+func MkdirAllChown(path string, perm os.FileMode, flags AtomicWriteFlags, uid sys.UserID, gid sys.GroupID) error {
```

Comment:
I think it's probably reasonable to add the flags argument here, as that's consistent with the `AtomicWrite` family of functions. However, the flags are different from `AtomicWriteFlags` (we only need the chmod flag here) so let's create a new type `MkdirFlags` with a `MkdirChmod` flag.